fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt

```diff
@@ -0,0 +1,248 @@
+    ): View {
+        _binding = FragmentHomeBinding.inflate(inflater, container, false)
+
+        val activity = activity as HomeActivity
+
+        onboardingAccountObserver = OnboardingAccountObserver(
+            context = requireContext(),
+            dispatchChanges = ::dispatchOnboardingStateChanges,
+        )
+
+        onboardingStore = StoreProvider.get(this) {
+            OnboardingStore(
+                initialState = OnboardingFragmentState(
+                    onboardingState = onboardingAccountObserver.getOnboardingState(),
+                ),
+            )
+        }
+
+        _interactor = DefaultOnboardingInteractor(
+            controller = DefaultOnboardingController(
+                activity = activity,
+                navController = findNavController(),
+                hideOnboarding = ::hideOnboarding,
+            ),
+            privateBrowsingController = DefaultPrivateBrowsingController(
+                activity = activity,
+                appStore = requireComponents.appStore,
+                navController = findNavController(),
+            ),
+            searchSelectorController = DefaultSearchSelectorController(
+                activity = activity,
+                navController = findNavController(),
+            ),
+            toolbarController = DefaultToolbarController(
+                activity = activity,
+                store = requireComponents.core.store,
+                navController = findNavController(),
+            ),
+        )
+
+        toolbarView = ToolbarView(
+            binding = binding,
+            context = requireContext(),
+            interactor = interactor,
+        )
+
+        onboardingView = OnboardingView(
+            containerView = binding.sessionControlRecyclerView,
+            interactor = interactor,
+        )
+
+        activity.themeManager.applyStatusBarTheme(activity)
+
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        consumeFrom(onboardingStore) { state ->
+            onboardingView?.update(state.onboardingState, requireComponents.fenixOnboarding.config)
+        }
+
+        homeMenuView = HomeMenuView(
+            view = view,
+            context = view.context,
+            lifecycleOwner = viewLifecycleOwner,
+            homeActivity = activity as HomeActivity,
+            navController = findNavController(),
+            menuButton = WeakReference(binding.menuButton),
+            hideOnboardingIfNeeded = interactor::onStartBrowsingClicked,
+        ).also { it.build() }
+
+        tabCounterView = TabCounterView(
+            context = requireContext(),
+            browsingModeManager = browsingModeManager,
+            navController = findNavController(),
+            tabCounter = binding.tabButton,
+        )
+
+        toolbarView?.build()
+
+        PrivateBrowsingButtonView(
```

Comment:
do we need to keep this as a class variable to clean it up like the others?

Reply:
I don't think so because we don't need to keep a reference to it for any type of `update` in other places.